analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -235,54 +227,57 @@ class GoMod(
     }
 }
 
-private data class Edge(
-    val source: Identifier,
-    val target: Identifier
-)
-
-private fun Collection<Edge>.getNodes(): Set<Identifier> = flatMap { listOf(it.source, it.target) }.toSet()
-
-private fun Collection<Edge>.toPackageReferenceForest(
-    ignoreNode: (Identifier) -> Boolean = { false }
-): SortedSet<PackageReference> {
-    data class Node(
-        val id: Identifier,
-        val outgoingEdges: MutableSet<Identifier> = mutableSetOf(),
-        val incomingEdges: MutableSet<Identifier> = mutableSetOf()
-    )
+/**
+ * A type alias for a representation of the dependency graph used by this class. This is basically an adjacency list
+ * implemented by a map whose keys are package identifiers and whose values are the identifiers of packages these
+ * packages depend on.
+ */
+private typealias Graph = MutableMap<Identifier, Set<Identifier>>
```

Comment:
Instead of defining a `typealias`, I believe it is worth to define a `class`.
This allows to define all the below extension functions simply as member functions,
provides better encapsulation and is simpler to use.

Reply:
`Graph` is now a class.